### DISCUSSION TITLE
Specific filter for custom link

### DISCUSSION THREAD

**Discussion by tux-mania** [netbox-community/netbox#11024] [Q&A] [answered]
Hello all!

I need some help please in order to retrieve some data.
On a custom link (displayed on the site), I need to get data from one of the circuit assigned to it.
I don't know yet how to do it on the Netbox' UI side, but I'm still trying to find it with the Netbox shell.

The way I tried (which is not working):
```
>>> circuit = Site.objects.get(name='SITE-NAME').circuit_terminations.get(circuit__custom_field_data__WAN='2')
>>> circuit
<CircuitTermination: Termination A: SITE-NAME>
>>> circuit.custom_field_data.get('IP')
>>>
```

Obviously, the custom field 'IP' is not empty, but I am not able to retrieve the data.

Could someone help me on this one please?
Thanks a lot for your time.

**Comment by candlerb** [comment, marked as answer]
Just to state the obvious, you searched for a circuit with `custom_field_data["WAN"]`, but you are reading `custom_field_data["IP"]`.

What value do you get from the expression `circuit.custom_field_data`, without the `get('IP')` on the end?

**Reply by tux-mania** [reply]
Sorry if it wasn't well explained from the beginning... To be more specific:
I have a site with 3 circuits assigned to it. Each circuit has custom fields "WAN" and "IP". I have WAN=1 for 1rst circuit, WAN=2 for the second, etc.
I need to get the value of the custom field "IP" on the circuit assigned to the site which has the custom field "WAN" = 2 

I get the following result:
```
>>> circuit = Site.objects.get(name='SITE-NAME').circuit_terminations.get(circuit__custom_field_data__WAN='2').custom_field_data
>>> circuit
{}
```

**Reply by candlerb** [reply]
What does this show?

```
>>> cterm = Site.objects.get(name='SITE-NAME').circuit_terminations.get(circuit__custom_field_data__WAN='2')
>>> type(cterm)
```

It seems to me that you're looking for custom field data on the *circuit termination*, but you've only checked whether custom_field_data exists on the *circuit*.

What makes you think that there is custom field data on the circuit termination?

Maybe what you're looking for is:

```
>>> cterm.circuit.custom_field_data
```

**Reply by tux-mania** [reply]
Hi @candlerb . Thank you for your help.
The `cterm.circuit.custom_field_data` was exactly what I was looking for! I still get lost with all of these syntax...

Thanks again for your help and your time!!!

**Reply by candlerb** [reply]
No problem.  You just have to remember that *circuit* and *circuit termination* are two different objects, which are linked together:

```
circuit ------< circuit termination
```

i.e. each circuit termination "belongs to" a particular circuit.

Hence while you can set custom_field_data on circuit, you can also set it on the related circuit terminations as well.
